src/mmw/apps/modeling/mapshed/tasks.py

```diff
@@ -304,6 +318,132 @@ def avg_awc(sjs_result):
     }
 
 
+@shared_task(throws=Exception)
+def nlcd_slope(result):
+    if 'error' in result:
+        raise Exception('[nlcd_slope] {}'.format(result['error']))
+
+    result = parse_sjs_result(result)
+
+    ag_slope_3_count = 0
+    ag_slope_3_8_count = 0
+    ag_count = 0
+    total_count = 0
+
+    for (nlcd_code, slope), count in result.iteritems():
+        if nlcd_code in AG_NLCD_CODES:
+            if slope > 3:
+                ag_slope_3_count += count
+            if 3 < slope < 8:
+                ag_slope_3_8_count += count
+            ag_count += count
+
+        total_count += count
+
+    # percent of AOI that is agricultural with slope > 3%
+    # see Class1.vb#7223
+    ag_slope_3_pct = (float(ag_slope_3_count) / total_count
+                      if total_count > 0 else 0.0)
+
+    # percent of AOI that is agricultural with 3% < slope < 8%
+    ag_slope_3_8_pct = (float(ag_slope_3_8_count) / total_count
+                        if total_count > 0 else 0.0)
+
+    # percent of agricultural parts of AOI with slope > 3%
+    # see Class1.vb#9864
+    n41 = float(ag_slope_3_count) / ag_count if ag_count > 0 else 0.0
+
+    output = {
+        'ag_slope_3_pct': ag_slope_3_pct,
+        'ag_slope_3_8_pct': ag_slope_3_8_pct,
+        'n41': n41
+    }
+
+    # TODO remove before merging
+    print(output)
+
+    return output
+
+
+@shared_task(throws=Exception)
+def slope(result):
+    if 'error' in result:
+        raise Exception('[slope] {}'.format(result['error']))
+
+    result = parse_sjs_result(result)
+
+    # average slope over the AOI
+    # see Class1.vb#6252
+    avg_slope = result[0]
+
+    output = {
+        'avg_slope': avg_slope
+    }
+
+    # TODO remove before merging
+    print(output)
+
+    return output
+
+
+def get_lu_index(nlcd):
+    if nlcd == 81:
+        lu_index = 1
+    elif nlcd == 82:
+        lu_index = 2
+    elif nlcd in [41, 42, 43, 52]:
+        lu_index = 3
+    elif nlcd in [90, 95]:
+        lu_index = 4
+    elif nlcd in [21, 71]:
+        lu_index = 7
+    elif nlcd in [12, 31]:
+        lu_index = 8
+    elif nlcd == 22:
+        lu_index = 11
+    elif nlcd == 23:
+        lu_index = 12
+    elif nlcd == 24:
+        lu_index = 13
+    else:
+        return None
+
+    return lu_index - 1
+
+
+@shared_task(throws=Exception)
+def nlcd_kfactor(result):
+    if 'error' in result:
+        raise Exception('[nlcd_kfactor] {}'.format(result['error']))
+
+    result = parse_sjs_result(result)
+
+    # average kfactor for each land use
+    # see Class1.vb#6431
+    kf = [0.0] * NLU
+    for nlcd_code, kfactor in result.iteritems():
+        lu_ind = get_lu_index(nlcd_code)
+        if lu_ind:
```

Comment:
There's a bug on this line. `lu_ind` is correctly `0` for `nlcd_code == 81` but `if lu_ind` considers it `False`. Switch to using `if lu_ind != None` or something more specific.

```python
result == {
    "21": 0.31612825878492506,
    "22": 0.290272814376818,
    "23": 0.2710281603574055,
    "24": 0.24679893810064235,
    "31": 0.2552305324809915,
    "41": 0.32789170498527764,
    "42": 0.3459407571914061,
    "43": 0.3389575358318246,
    "52": 0.327332149092242,
    "71": 0.30504905618239603,
    "81": 0.3226653785731504,
    "82": 0.32048570821643807,
    "90": 0.3332366971914464,
    "95": 0.2594399942085147
}

kf == [
    0,
    0.32048570821643807,
    0.327332149092242,
    0.2594399942085147,
    0,
    0,
    0.31612825878492506,
    0.2552305324809915,
    0,
    0,
    0.290272814376818,
    0.2710281603574055,
    0.24679893810064235,
    0,
    0,
    0
]
```